support-services/src/test/scala/com/gu/support/redemption/generator/GiftCodeGeneratorSpec.scala

```diff
@@ -7,13 +7,13 @@ class GiftCodeGeneratorSpec extends AnyFlatSpec with Matchers {
 
   it should "work in the basic case" in {
     val giftCode = GiftCodeGenerator.fromInts(Iterator.continually(0)).next.withDuration(GiftDuration.Gift3Month)
-    giftCode.value should be("gd03-000000")
+    giftCode.value should be("gd03-00000000")
   }
 
   it should "always produce different codes with the built in random" in {
     val generateGiftCode = GiftCodeGenerator.randomGiftCodes
     val giftCodes = generateGiftCode.map(_.withDuration(GiftDuration.Gift3Month))
-    val numberToCheck = 10000
+    val numberToCheck = 100000
```

Comment:
is it a lot slower? I was thinking when I wrote it even 10000 was probably excessive, even 100 is probably plenty to catch the obvious errors

Reply:
Around 897ms - seems ok to me?

Reply:
1 second for such an unlikely test to fail seems disproportionate, unit tests are generally very quick. I would recommend reducing it as it's actually just designed to find major breakages e.g. total loss of randomness.